tests/unit/converter/test_biosamples_converter.py

```diff
@@ -7,76 +7,201 @@
-
+        # then
         self.assertEqual(SampleMatcher(biosample), converted_bio_sample)
 
     def test_given_ingest_biomaterial_when_release_date_not_defined_in_project_converts_correct_biosample(
             self):
+        # given
         biomaterial = self.__get_biomaterial_by_index(0)
+        biosample = self.__create_a_sample()
+        # when
+        converted_bio_sample = self.biosamples_converter.convert(biomaterial)
+        # then
+        self.assertEqual(SampleMatcher(biosample), converted_bio_sample)
 
-        submission_date = '2019-07-18T21:12:39.770Z'
-        additional_attributes = {
-            'domain': self.domain
-        }
-
-        biosample = self.__create_a_sample(submission_date)
-
-        converted_bio_sample = self.biosamples_converter.convert(biomaterial, additional_attributes )
-
+    def test_when_biomaterial_has_no_name_converted_biosample_has_biomaterial_id_as_name(self):
+        # given
+        biomaterial = self.__get_biomaterial_by_index(1)
+        biosample = self.__create_a_sample(biomaterial_id='BP37d', name='BP37d')
+        # when
+        converted_bio_sample = self.biosamples_converter.convert(biomaterial)
+        # then
         self.assertEqual(SampleMatcher(biosample), converted_bio_sample)
 
-    def __create_a_sample(self, release_date):
-        biosample = Sample(
+    def test_sample_can_convert_sample_with_no_attributes(self):
+        # given
+        biomaterial = self.__get_biomaterial_by_index(0)
+        target_biosample = self.__create_a_sample()
+        # when
+        converted_biosample = self.biosamples_converter.convert(biomaterial)
+        # then
+        self.assertEqual(SampleMatcher(target_biosample), converted_biosample)
+
+    def test_sample_throws_error_with_no_domain(self):
+        # given
+        self.biosamples_converter = BioSamplesConverter()
+        biomaterial = {}
+        # then
+        with self.assertRaises(MissingBioSamplesDomain):
+            # when
+            self.biosamples_converter.convert(biomaterial)
+
+    def test_sample_throws_error_with_no_name(self):
+        # given
+        biomaterial = {}
+        # then
+        with self.assertRaises(MissingBioSamplesSampleName):
+            # when
+            self.biosamples_converter.convert(biomaterial)
+
+    def test_sample_uses_attribute_accession(self):
+        # given
+        biomaterial = self.__get_biomaterial_by_id('accession-attribute-no-content-test')
+        target_biosample = self.__create_a_sample()
+        # when
+        additional_attributes = {
+            'accession': biomaterial.get('accession')
+        }
+        converted_biosample = self.biosamples_converter.convert(biomaterial, additional_attributes)
+        # then
+        self.assertEqual(SampleMatcher(target_biosample), converted_biosample)
+
+    def test_sample_uses_attribute_accession_over_core_accession(self):
```

Comment:
What is the difference between this test and `test_sample_uses_attribute_accession`?
As I checked the code it is the very same test.

Reply:
I'll change the names to make the difference clearer.